### PR TITLE
aliakbar/added_p2p_ping_api_call

### PR DESCRIPTION
**Pull request by aliakbar-deriv**

### CODE REVIEW

File: lib/api/response/p2p_ping_response_result.dart [hunk 1]
```diff
@@ -44,6 +50,40 @@ class P2pPingResponse extends P2pPingResponseModel {
     return resultMap;
   }
 
+  static final BaseAPI _api = Injector()<BaseAPI>();
```
Comment (ramin-deriv) [inline lib/api/response/p2p_ping_response_result.dart:53]: @aliakbar-deriv Are these generated by the code generator which is described in `flutter-deriv-api` docs?
https://github.com/deriv-com/flutter-deriv-api/tree/dev

Comment (hamed-deriv) [reply lib/api/response/p2p_ping_response_result.dart:53]: @ramin-deriv this part is not completely auto-generated.

@aliakbar-deriv please add a JSON file to `basic_api/generated/methods` to make it automated next time.
you can take a look at this file as an example:
app_get_receive_methods.json

Comment (aliakbar-deriv) [reply lib/api/response/p2p_ping_response_result.dart:53]: @ramin-deriv @hamed-deriv This was not auto-generated. The file `lib/basic_api/generated/methods/p2p_ping_receive_methods.json` is already there. Let me try update this file and auto-generate it. As of now I checked out this branch from `dev`. From which branch should I checkout?

Comment (hamed-deriv) [reply lib/api/response/p2p_ping_response_result.dart:53]: @aliakbar-deriv `dev` branch is correct if you are not working with Flutter 3.
you need to update `p2p_ping_receive_methods.json` manually for the first time, after that class methods will be generated based on the JSON file.

Comment (aliakbar-deriv) [reply lib/api/response/p2p_ping_response_result.dart:53]: Understood! Thanks, Hamed.

File: lib/api/exceptions/p2p_ping_exception.dart [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import 'common_exception.dart';
+
+/// P2P Ping exception class
+class P2pPingException extends CommonException {
+  /// Initializes
+  P2pPingException({
+    BaseExceptionModel? baseExceptionModel,
+  }) : super(baseExceptionModel: baseExceptionModel);
+}
```
Comment (hamed-deriv) [inline lib/api/exceptions/p2p_ping_exception.dart:11]: you can remove this class.

Comment (aliakbar-deriv) [reply lib/api/exceptions/p2p_ping_exception.dart]: Removed it and replaced with 'APIBaseException'.

**Review by ramin-deriv — APPROVED** [review]

**Review by hamed-deriv — APPROVED** [review]

**Pull request merged**